org.lflang/src/org/lflang/generator/DockerGeneratorBase.java

```diff
@@ -2,18 +2,33 @@
 
 import java.io.IOException;
 import java.nio.file.Path;
-import java.util.HashMap;
-import java.util.Map;
-import org.lflang.FileConfig;
-import org.lflang.TargetConfig;
+import java.util.ArrayList;
+import java.util.List;
 import org.lflang.util.FileUtil;
 
 /**
  * The base class for docker file related code generation.
  *
+ * The design of abstractions is as follows:
+ *
+ * Docker-facing API
+ * - This ("DockerGeneratorBase") class defines a "DockerData" class
+ *   that specifies the information it needs to generate docker files and
+ *   docker compose files for any target. This is the docker-facing
+ *   API.
+ *
+ * Target Code Generator-facing API
+ * - Each target-specific docker generator implements this class and
+ *   defines in themselves a class that implements the "GeneratorData"
+ *   interface of this class. This is the target code generator-facing API.
+ *
+ * The purpose of this abstraction design is to contain all
+ * docker-specific information inside docker generator classes and
+ * prevent docker-related information from polluting target code generators.
+ *
  * @author{Hou Seng Wong <housengw@berkeley.edu>}
  */
-public class DockerGeneratorBase {
+abstract public class DockerGeneratorBase {
```

Comment:
Are different implementations of this class expected to differ much?

Reply:
the main difference between different implementations is the way they populate the `DockerData` class based on `?GeneratorData` class, which is target language specific